### ISSUE TITLE
Issue with LambdaMart and Learning to Rank

### ISSUE THREAD

**Issue by etiennes92** [dmlc/xgboost#10388]
Hello I am trying to use LambdaMart with xgbRanker.
In the tutorial, the fit method takes the argument aid, as it should be:

ranker = xgb.XGBRanker(tree_method="hist", lambdarank_num_pair_per_sample=8, objective="rank:ndcg", lambdarank_pair_method="topk")
ranker.fit(X, y, qid=qid[sorted_idx])

But when it comes to prediction, there is any qid argument in the predict method in the tutorial:

scores = ranker.predict(X)
sorted_idx = np.argsort(scores)[::-1]
scores = scores[sorted_idx]

I don’t understand how the predict method is not taking into consideration a qid argument. Indeed, in my test set I could have multiple queries with several documents per query. But here it is all considered the same query.

Should I only use the predict method on each query independently on my test set ?

Thank you very much for your responses

**Comment by hcho3** [comment]
Yes, currently the predict method assumes that all documents in X belong to the same query. To handle multiple queries, you will need to slice the result of `predict(X)` according to the query boundary.

**Comment by trivialfis** [comment]
The query group is used to calculate the gradient for training. After the training, the model doesn't need to use the query group information and is the same for any query.

**Comment by trivialfis** [comment]
Feel free to reopen if there are further questions.
